src/other/metadata.jl

```diff
@@ -745,11 +722,14 @@ end
 # from Tables.jl table src to dst, discarding previous metadata contents of dst
 function _copy_all_all_metadata!(dst::DataFrame, src)
     emptymetadata!(dst)
-    for key in metadatakeys(src)
-        val, style = metadata(src, key, style=true)
-        metadata!(dst, key, val, style=style)
+    if DataAPI.metadatasupport(typeof(src)).read
+        for key in metadatakeys(src)
+            val, style = metadata(src, key, style=true)
+            metadata!(dst, key, val, style=style)
+        end
     end
     emptycolmetadata!(dst)
+    DataAPI.colmetadatasupport(typeof(src)).read || return nothing
```

Comment:
Use an `if` for consistency with above?

Reply:
OK. I have changed short-circut to `if`s everywhere then. After CI passes I will merge this PR.
Then I will wait one day to wait for final responses, and will make a release of DataFrames.jl 1.4 tomorrow if there are none.